ui/src/app/shared/components/revision.tsx

```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
-    revision = revision || '';
+export const Revision = ({repoUrl, revision, path, title, children}: {repoUrl: string; revision: string; path?: string; title?: string; children?: React.ReactNode}) => {
     const hasPath = path && path !== '.';
+    if (title == 'PATH' && path) {
+        return <span></span>
+    }
```

Comment:
```suggestion
    if (title == 'PATH' && !path) {
        // This source literally has no path, so we won't show one.
        return <span></span>
    }
    const hasPath = path && path !== '.';
```

Just to clean up.

Do we have to pass `title`?

Reply:
title is just to distinguish the field that is calling the Revision tag. The same Revision tag is getting used to return the value of TargetRevision which includes path in generating the URL. So, if we don't specify `title`, the value of `TargetRevision` also gets set empty.

Reply:
Ah, sure. Can we pass a boolean instead? `isForRevision: false`, or something like that?